sklearn/feature_extraction/heterogeneous.py

```diff
@@ -0,0 +1,148 @@
+from scipy import sparse
+import numpy as np
+
+from ..base import BaseEstimator, TransformerMixin
+from ..pipeline import _fit_one_transformer, _fit_transform_one, _transform_one
+from ..externals.joblib import Parallel, delayed
+from ..externals.six import iteritems
+
+
+class ColumnTransformer(BaseEstimator, TransformerMixin):
+    """Applies transformers to columns of a dataframe / dict.
+
+    This estimator applies transformer objects to columns or fields of the
+    input, then concatenates the results. This is useful for heterogeneous or
+    columnar data, to combine several feature extraction mechanisms into a
+    single transformer.
+
+    Read more in the :ref:`User Guide <column_transformer>`.
+
+    Parameters
+    ----------
+    transformers : dict from string to (string, transformer) tuples
```

Comment:
The implementation is expected the dict values to be `(transformer, string)` tuples, and not `(string, transformer)` as documented here.